src/ChatGptNet/IChatGptClient.cs

```diff
@@ -41,42 +42,45 @@ Task<Guid> SetupAsync(string message)
     /// <param name="message">The message.</param>
+    /// <param name="chatGptParameters">A <seealso cref="ChatGptParameters"/> object that it's used to override the default completion parameters in the <see cref="ChatGptOptions.DefaultModel"/></param>
     /// <param name="cancellationToken">A <see cref="CancellationToken"/> that can be used by other objects or threads to receive notice of cancellation.</param>
     /// <returns>The chat completion response.</returns>
     /// <exception cref="ArgumentNullException"><paramref name="message"/> is <see langword="null"/>.</exception>
     /// <exception cref="ChatGptException">An error occurred while calling the API and the <seealso cref="ChatGptOptions.ThrowExceptionOnError"/> is <see langword="true"/>.</exception>
     /// <seealso cref="ChatGptRequest"/>
     /// <seealso cref="ChatGptResponse"/>
     /// <seealso cref="ChatGptOptions"/>
-    Task<ChatGptResponse> AskAsync(Guid conversationId, string message, CancellationToken cancellationToken = default)
-        => AskAsync(conversationId, message, null, cancellationToken);
+    Task<ChatGptResponse> AskAsync(Guid conversationId, string message, ChatGptParameters? chatGptParameters = null, CancellationToken cancellationToken = default)
+        => AskAsync(conversationId, message, null, chatGptParameters, cancellationToken);
 
     /// <summary>
     /// Requests a chat interaction.
     /// </summary>
     /// <param name="conversationId">The unique identifier of the conversation, used to automatically retrieve previous messages in the chat history.</param>
     /// <param name="message">The message.</param>
     /// <param name="model">The chat completion model to use. If model is <see langword="null"/>, then the one specified in the <see cref="ChatGptOptions.DefaultModel"/> property will be used.</param>
+    /// <param name="chatGptParameters">A <seealso cref="ChatGptParameters"/> object that it's used to override the default completion parameters in the <see cref="ChatGptOptions.DefaultModel"/></param>
     /// <param name="cancellationToken">A <see cref="CancellationToken"/> that can be used by other objects or threads to receive notice of cancellation.</param>
     /// <returns>The chat completion response.</returns>
     /// <exception cref="ArgumentNullException"><paramref name="message"/> is <see langword="null"/>.</exception>
     /// <exception cref="ChatGptException">An error occurred while calling the API and the <see cref="ChatGptOptions.ThrowExceptionOnError"/> is <see langword="true"/>.</exception>
     /// <seealso cref="ChatGptRequest"/>
     /// <seealso cref="ChatGptResponse"/>
-    Task<ChatGptResponse> AskAsync(Guid conversationId, string message, string? model, CancellationToken cancellationToken = default);
+    Task<ChatGptResponse> AskAsync(Guid conversationId, string message, string? model, ChatGptParameters? chatGptParameters = null, CancellationToken cancellationToken = default);
```

Comment:
I think it is better to have _parameters_ argument before _model_ (in all the signatures where there are both arguments), because it's more likely that a user want to change parameters rather then the model name.